src/dev/impl/DevToys/ViewModels/Tools/Graphic/ImageConverter/ImageConverterToolViewModel.cs

```diff
@@ -0,0 +1,303 @@
+                    {
+                        filePicker.FileTypeFilter.Add(SupportedFileExtensions[i]);
+                    }
+
+                    IReadOnlyList<StorageFile> files = await filePicker.PickMultipleFilesAsync();
+                    for (int i = 0; i < files.Count; i++)
+                    {
+                        QueueNewConversionWorkItem(files[i]);
+                    }
+                });
+            }
+            catch (Exception ex)
+            {
+                Logger.LogFault(nameof(ImageConverterToolViewModel), ex, "Unable to browse file(s).");
+                IsInfoBarOpen = true;
+            }
+        }
+
+        #endregion
+
+        #region DeleteAllCommand
+
+        public IRelayCommand DeleteAllCommand { get; }
+
+        private void ExecuteDeleteAllCommand()
+        {
+            ConversionWorkQueue.Clear();
+        }
+
+        #endregion
+
+        #region SaveAllCommand
+
+        public IAsyncRelayCommand SaveAllCommand { get; }
+
+        private async Task ExecuteSaveAllCommandAsync()
+        {
+            try
+            {
+                var works = ConversionWorkQueue.ToList();
+
+                var folderPicker = new FolderPicker
+                {
+                    ViewMode = PickerViewMode.List
+                };
+
+                StorageFolder? selectedFolder = await folderPicker.PickSingleFolderAsync();
+
+                if (selectedFolder is not null)
+                {
+                    ICollection<Task> conversionTasks = new List<Task>();
+
+                    foreach (ImageConversionWorkItem work in works)
+                    {
+                        conversionTasks.Add(SaveConversionWorkItem(selectedFolder, work));
+                    }
+
+                    await Task.WhenAll(conversionTasks);
+                }
+            }
+            catch (Exception ex)
+            {
+                Logger.LogFault(nameof(ImageConverterToolViewModel), ex, "Unable to save all files.");
+                IsInfoBarOpen = true;
+            }
+        }
+
+        #endregion
+
+        #region CloseInfoBarButtonCommand
+
+        public IRelayCommand CloseInfoBarButtonCommand { get; }
+
+        private void ExecuteCloseInfoBarButtonCommand()
+        {
+            IsInfoBarOpen = false;
+        }
+
+        #endregion
+
+        private async Task SaveConversionWorkItem(StorageFolder selectedFolder, ImageConversionWorkItem work)
+        {
+            StorageFile newFile = await selectedFolder.CreateFileAsync(string.Concat(work.File.DisplayName, ImageHelper.GetExtension(ConvertedFormat)), CreationCollisionOption.ReplaceExisting);
+            using (IRandomAccessStream outputStream = await newFile.OpenAsync(FileAccessMode.ReadWrite))
+            {
+                BitmapEncoder? encoder = await ImageHelper.GetEncoderAsync(ConvertedFormat, outputStream);
+                var bitmap = await work.DecodeImageAsync();
```

Comment:
Here too, please don't forget to dispose this resource after `FlushAsync`.

Reply:
I was about to push it 😝

Reply:
Hahaha, sorry, I've been too fast then :-)